guides/documentations/essentials.md

```diff
@@ -0,0 +1,265 @@
+DefaultRender.js,                // Class that compiles a Component to data.
+DefaultParser.js,                // Class used for parsing data to Component.
+DefaultDrawer.js,                // Class used for drawing a Component in a graphical representation.
+DefaultData.js,                  // Class that represents all data of a Plugin.
+DefaultConfiguration.js,         // Class that represents default configuration of a Plugin.
+DefaultPlugin.js,                // Class that represents the default Plugin structure.
+Component.js,                    // Class that represents the base of modeling tools.
+ComponentDefinition.js,          // Class used for the definition of Component's datas and constraints.
+ComponentAttribute.js,           // Class used to add Attribute in Components.
+ComponentAttributeDefinition.js, // Class used to define Attributes for type.
+ComponentDrawOption.js,          // Class that represents option for the Component drawing.
+ComponentLink.js,                // Class that links Components together.
+ComponentLinkDefinition.js,      // Class used for the definition of the link between Components.
+ParseError.js,                   // Class that represents a parsing error. Used by the DefaultParser as default Error.
+FileInformation.js,              // Class that represents the object to store all file information.
+FileInput.js,                    // Class that represents the object to store the file content.
+Tag.js,                          // Class that represents a tag in Leto-modelizer. Used by the DefaultConfiguration class.
+Variable.js,                     // Class that represents a variable in a file of a diagram. Used by the DefaultData class.
+```
+
+| Plugin lifecycle |
+| :--: |
+| <img height="400" src="../../documentations/lifecycle.svg"/> |
+
+This is the default lifecycle of plugin usage in Leto Modelizer.
+
+## Metadata
+
+To make your plugin work, we need to have an array of [ComponentDefinitions](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/src/models/ComponentDefinition.js).
+Component definitions are loaded from a metadata file. Each Component definition has:
+- A name
+- An icon
+- Attributes
+- ...
+
+| Left: Components in Leto-modelizer - Right: Related Metadata |
+| :--: |
+| <img height="400" src="../../documentations/components.png"/> |
+
+Attributes of a Component represent all the related properties that can/must be set.
+Link Definitions can also be loaded from the metadata file (for colored links, dashed links...).
+
+Once the Component definitions are loaded, components can be instantiated from the "components palette" in Leto-modelizer.
+The DefaultMetadata provides two methods:
+
+- `validate`: Validate the metadata.
+- `parse`: Set all component/link definitions in PluginData from your provided metadata.
+
+## Renderer / Parser
+
+| Parse and Render |
+| :--: |
+| <img height="400" src="../../documentations/parser_renderer.png"/> |
```

Comment:
not a huge fan of this image because is not valid:
- renderer generate components from source files and put it in storage
- parser generate source files from components in storage
- drawer generate diagram from components in storage

Reply:
added a Summary part with image